### ISSUE TITLE
objectMode=true by default in v4

### ISSUE THREAD

**Issue by kafkahw** [adaltas/node-csv-stringify#75]
1. anybody know why `objectMode` is set to true by default for the transform stream returned by `stringify`?

2. any plan to support for `objectMode` in `options` of `stringify` so that we can override the default value? Reason: some server (e.g. `hapijs`) only supports `stream` response if the stream is not `objectMode`.

**Comment by wdavidw** [comment]
The reason why we set `objectMode` to true is that the writable stream expects objects as input. While I was recently [re-writing the CSV parser](http://www.adaltas.com/en/2018/11/19/nodejs-csv-version-4-performance/), I realized there are now two properties `readableObjectMode` and `writableObjectMode` to dissociate input from output. I have just pushed the [modification](94cc6f384888cb65ee0383944cb5f11aa9c2d52c) on the master branch.

Sorry, I can't make a release right now, there are some minor breaking changes in the pipe (options exposed in underscore form) and I wish to add a few more before releasing a major version (mostly renaming the options `row_delimiter` to `record_delimiter` and `formatters` to `cast` to be aligned with the parser API).
